contracts/vaults/BaseVaults/RibbonThetaVault.sol

```diff
@@ -395,14 +406,29 @@ contract RibbonThetaVault is RibbonVault, RibbonThetaVaultStorage {
      * @notice Rolls the vault's funds into a new short position.
      */
     function rollToNextOption() external onlyKeeper nonReentrant {
+        uint256 currQueuedWithdrawShares = currentQueuedWithdrawShares;
+
         (
             address newOption,
             uint256 lockedBalance,
             uint256 queuedWithdrawAmount
-        ) = _rollToNextOption(uint256(lastQueuedWithdrawAmount));
+        ) =
+            _rollToNextOption(
+                lastQueuedWithdrawAmount,
+                currQueuedWithdrawShares
+            );
 
         lastQueuedWithdrawAmount = queuedWithdrawAmount;
 
+        uint256 newQueuedWithdrawShares =
+            uint256(vaultState.queuedWithdrawShares).add(
+                currQueuedWithdrawShares
+            );
+        ShareMath.assertUint128(newQueuedWithdrawShares);
+        vaultState.queuedWithdrawShares = uint128(newQueuedWithdrawShares);
+
+        currentQueuedWithdrawShares = 0;
```

Comment:
The `vaultState.queuedWithdrawShares` value is used in `_rollToNextOption`

The rollover logic would break if we modify that variable before rolling over

I think we can set `currentQueuedWithdrawShares = 0;` after rolling over since it's easier to reason about the state changes if they happen all at once.
If we set it after the assignment `currQueuedWithdrawShares = currentQueuedWithdrawShares;` then we would have some state changes before rolling over and some after which would make the function harder to reason about.